timescaledb/tutorials/analyze-intraday-stocks/explore-stocks-data.md

```diff
@@ -0,0 +1,286 @@
+
+Let's see how it's done!
+
+## Install Plotly and Pandas
+
+In order to get started with data exploration, you need to install a couple of tools first:
+
+* Pandas, to query and structure the data
+* Plotly, to create visualizations quickly
+
+**Install both**
+```bash
+pip install plotly pandas
+```
+
+Then, you will need to either open a new python file or use a Jupyter notebook where you can 
+start exploring the dataset.
+
+## Establish database connection
+
+Use the config file you created before and psycopg2 to create a database connection object.
+
+```python
+import config, psycopg2
+conn = psycopg2.connect(database=config.DB_NAME,
+                        host=config.DB_HOST,
+                        user=config.DB_USER,
+                        password=config.DB_PASS,
+                        port=config.DB_PORT)
+```
+
+In each data exploration script below, you will need to reference this connection object to be able to 
+query the database.
+
+## Generate stock market insights
+
+Let's start off analyzing trading volumes, then have a look at weekly price points, and finally, dig deep on 
+price changes.
+
+<highlight type="tip">
+Let these queries serve as inspiration to you, and feel free to change things up, like the analyzed *time frame*, the *symbol* or other parts of the query. Have fun!
+</highlight>
+
+1. Which symbols have the highest transaction volumes?
+2. How did Apple's trading volume change over time?
+3. How did Apple's stock price change over time?
+4. Which symbols had the highest weekly gains?
+5. Weekly FAANG prices over time?
+6. Weekly price changes of Apple, Facebook, Google?
+7. Distribution of daily price changes of Amazon and Zoom
+8.  Apple 15-min candlestick chart
+
+### 1. Which symbols have the highest transaction volumes?
+
+Let's generate a bar chart which shows the most traded symbols in the last 14 days.
+
+```python
+query = """
+    SELECT symbol, sum(trading_volume) AS volume
+    FROM stocks_intraday
+    WHERE (now() - date(time)) < INTERVAL '{time_frame}'
+    GROUP BY symbol
+    ORDER BY volume DESC
+    LIMIT 5
+""".format(time_frame="14 day")
+df = pd.read_sql(query, conn)
+fig = px.bar(df, x='symbol', y='volume', title="Most traded symbols in the last 14 days")
+fig.show()
+```
+
+![most traded symbols](https://assets.timescale.com/docs/images/tutorials/intraday-stock-analysis/most_traded_symbols.png)
+
+### 2. How did Apple's trading volume change over time?
+
+Now let's try a similar query focused on the daily trading volume of one symbol (e.g. 'AAPL'). 
+
+```python
+query = """
+    SELECT time_bucket('{time_frame}', time) AS time_frame, sum(trading_volume) AS volume
```

Comment:
I honestly can't think of a different term ATM, but "time_frame" seems like the wrong term here. In most examples we use something like "bucket", although I'm not sure that's the best either. If anything, this is the _start_ of the bucket... so it's not really a time frame (start and end.... more like a range). 

Anyone have thoughts? (and whatever we decide should be consistent throughout)

Reply:
TBH, I just started using time_frame at first then sticked to it... but if we use bucket elsewhere, it should be used here as well imo

Reply:
https://github.com/timescale/docs/pull/150/commits/7c22159becad063bcb4ab076627c9f7b5b4e07c4